DataStore2/init.lua

```diff
@@ -182,15 +182,28 @@ end
 function DataStore:SetValidator(validator)
 	assert(
 		type(validator) == "function",
-		"function expected, got "..type(validator)
+		"function expected, got " .. type(validator)
 	)
 	self.validator = validator
 end
 
+local function AssertValidatorWithDefaultError(validator, input, defaultError)
+    local success, err = pcall(validator, input)
```

Comment:
I don't like this. If the validator fails, then it should error, full stop. I don't want to hide errors like this.